File: objc-sample-app/CoffeeListTableViewController.m
//
//  CoffeeListTableViewController.m
//  objc-sample-app
//
//  Created by Tapash Majumder on 6/21/18.
//  Copyright © 2018 Iterable. All rights reserved.
//

#import "CoffeeListTableViewController.h"
#import "CoffeeType.h"

@interface CoffeeListTableViewController ()
@end

@implementation CoffeeListTableViewController

- (void)setSearchTerm:(NSString *)searchTerm {
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self initializeCoffees];
    
    searchController = [[UISearchController alloc] initWithSearchResultsController:nil];
    searchController.searchBar.placeholder = @"Search";
    self.navigationItem.searchController = searchController;
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return coffees.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"coffeeCell" forIndexPath:indexPath];

    CoffeeType *coffee = coffees[indexPath.row];
    cell.textLabel.text = coffee.name;
    cell.imageView.image = coffee.image;
    
    return cell;
}


#pragma mark - Navigation

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
}

#pragma mark - private
UISearchController *searchController;

NSArray *coffees;

- (void) initializeCoffees {
    CoffeeType *cappuccino = [[CoffeeType alloc] initWithName:@"Cappuccino" andImage: [UIImage imageNamed:@"Cappuccino"]];
    CoffeeType *latte = [[CoffeeType alloc] initWithName:@"Latte" andImage: [UIImage imageNamed:@"Latte"]];
    CoffeeType *mocha = [[CoffeeType alloc] initWithName:@"Mocha" andImage: [UIImage imageNamed:@"Mocha"]];
    CoffeeType *black = [[CoffeeType alloc] initWithName:@"Black" andImage: [UIImage imageNamed:@"Black"]];

    coffees = @[cappuccino, latte, mocha, black];
}

@end

File: swift-sample-app/CoffeeListTableViewController.swift
//
//  CoffeeListTableViewController.swift
//  swift-sample-app
//
//  Created by Tapash Majumder on 6/15/18.
//  Copyright © 2018 Iterable. All rights reserved.
//

import UIKit

import IterableSDK

class CoffeeListTableViewController: UITableViewController {
    struct CoffeeType {
        let name: String
        let image: UIImage
        
        static let cappuccino = CoffeeType(name: "Cappuccino",image: #imageLiteral(resourceName: "Cappuccino"))
        static let latte = CoffeeType(name: "Latte", image: #imageLiteral(resourceName: "Latte"))
        static let mocha = CoffeeType(name: "Mocha", image: #imageLiteral(resourceName: "Mocha"))
        static let black = CoffeeType(name: "Black", image: #imageLiteral(resourceName: "Black"))
    }
    
    private let coffees: [CoffeeType] = [
        .cappuccino,
        .latte,
        .mocha,
        .black
    ]

    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    // MARK: - Table view data source
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return coffees.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "coffeeCell", for: indexPath)
        
        let coffee = coffees[indexPath.row]
        cell.textLabel?.text = coffee.name
        cell.imageView?.image = coffee.image
        
        return cell
    }
}
